### PR TITLE
feat: Perpetual link add light or dark mode

### PR DESCRIPTION
**Pull request by ChefMomota**

### CODE REVIEW

File: src/hooks/useTheme.ts [hunk 1]
```diff
@@ -1,11 +1,26 @@
...
 import { ThemeContext as StyledThemeContext } from 'styled-components'
 import { useTheme as useNextTheme } from 'next-themes'
 
+const COOKIE_THEME_KEY = 'theme'
+
 const useTheme = () => {
   const { resolvedTheme, setTheme } = useNextTheme()
   const theme = useContext(StyledThemeContext)
-  return { isDark: resolvedTheme === 'dark', theme, setTheme }
+  const getThemeCookie = Cookie.get(COOKIE_THEME_KEY)
```
Comment (memoyil) [inline src/hooks/useTheme.ts:11]: why do we need cookie? afaik we have theme property in local storage

Comment (ChefMomota) [reply src/hooks/useTheme.ts:11]: Because when click perpetual button, the theme will not same as our side.

Comment (memoyil) [reply src/hooks/useTheme.ts:11]: 👍

File: src/hooks/useTheme.ts [hunk 2]
```diff
@@ -1,11 +1,26 @@
...
+
+  if (!getThemeCookie) {
+    const themeValue = theme.isDark ? 'dark' : 'light'
+    Cookie.set(COOKIE_THEME_KEY, themeValue)
+  }
+
+  const handleSwitchTheme = (themeValue: 'light' | 'dark') => {
+    setTheme(themeValue)
+    Cookie.set(COOKIE_THEME_KEY, themeValue)
```
Comment (chefilip) [inline src/hooks/useTheme.ts:20]: Could you add error handle here in case cookie is full?

Comment (ChefMomota) [reply src/hooks/useTheme.ts]: Done

**Review by chefilip — APPROVED** [review]

**Pull request merged**